velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -865,6 +867,69 @@ RowVectorPtr VectorFuzzer::fuzzRowChildrenToLazy(RowVectorPtr rowVector) {
       std::move(children));
 }
 
+// Utility function to check if a RowVector can have nested lazy children.
+bool canWrapChildrenInLazy(const VectorPtr& baseVector) {
+  if (baseVector->typeKind() == TypeKind::ROW) {
+    if (VectorEncoding::isDictionary(baseVector->encoding())) {
+      return canWrapChildrenInLazy(baseVector->valueVector());
+    }
+    RowVector* rowVector = baseVector->as<RowVector>();
+    if (rowVector) {
+      return rowVector->nulls() == nullptr;
+    }
+  }
+
+  return false;
+}
+
+// Utility function That only takes a row vector that passes the check in
+// canWrapChildrenInLazy() and either picks the first child to be wrapped in
+// lazy OR picks the first row vector child that passes canWrapChildrenInLazy()
+// and traverses its children recursively.
+VectorPtr wrapInnermostChildToLazy(VectorPtr& baseVector) {
+  if (VectorEncoding::isDictionary(baseVector->encoding())) {
+    auto indices = baseVector->wrapInfo();
+    auto values = baseVector->valueVector();
+    auto nulls = baseVector->nulls();
+
+    auto copiedNulls = AlignedBuffer::copy(baseVector->pool(), nulls);
+
+    return BaseVector::wrapInDictionary(
+        copiedNulls,
+        indices,
+        baseVector->size(),
+        wrapInnermostChildToLazy(values));
+  }
+
+  RowVector* rowVector = baseVector->as<RowVector>();
+  VELOX_CHECK_NOT_NULL(rowVector);
+  std::vector<VectorPtr> children;
+  bool foundChildRow = false;
+  for (column_index_t i = 0; i < rowVector->childrenSize(); i++) {
+    auto child = rowVector->childAt(i);
+    if (!foundChildRow && canWrapChildrenInLazy(child)) {
+      child = wrapInnermostChildToLazy(child);
+      foundChildRow = true;
```

Comment:
nit: What about "haveWrapped"? The meaning of the name "foundChildRow" is not very clear to me.